File: src/main/python/net/x3djsonld/data/t3.py
###############################################
#
# Now available: developmental python x3d.py package on PyPi for import.
#   This approach greatly simplifies Python X3D deployment and use.
#   https://pypi.org/project/x3d
#
# Installation:
#       pip install x3d
# or
#       python -m pip install x3d
#
# Developer options for loading x3d package:
#
#    from x3d import *  # preferred approach, terser source that avoids x3d.* class prefixes
#
# or
#    import x3d         # traditional way to subclass x3d package, all classes require x3d.* prefix
#                       # but python source is very verbose, for example x3d.Material x3d.Shape etc.
#                       # X3dToPython.xslt stylesheet insertPackagePrefix=true supports this option.

from x3d import *

###############################################

newModel=X3D(profile='Immersive',version='3.3',
  head=head(
    children=[
    component(name='Networking',level=2),
    component(name='Core',level=2),
    meta(name='title',content='t3.x3d'),
    meta(name='identifier',content='http://coderextreme.net/X3DJSONLD/src/main/data/t4.x3d'),
    meta(name='generator',content='view3dscene, https://castle-engine.io/view3dscene.php'),
    meta(name='creator',content='Andreas Plesch and John Carlson'),
    meta(name='source',content='t1.wrl'),
    meta(name='description',content='Test Case for Proto Expander'),
    meta(content='https://www.web3d.org/x3d/content/examples/license.html',name='license'),
    meta(name='translated',content='12 May 2020'),
    meta(name='generator',content='DOM2JSONSerializer.js, https://github.com/coderextreme/X3DJSONLD/blob/master/src/main/node/DOM2JSONSerializer.js'),
    meta(name='reference',content='X3D JSON encoding: https://www.web3d.org/wiki/index.php/X3D_JSON_Encoding')]),
  Scene=Scene(
    children=[
    NavigationInfo(type='"EXAMINE" "FLY" "WALK"',speed=3,avatarSize=[200,200,120]),
    WorldInfo(title='Arts Mapper'),
    Viewpoint(description='looking North',position=(0,60,110),orientation=(1,0,0,-0.699999988079071),fieldOfView=0.785398125648499),
    Transform(translation=(-468,0,315),
      children=[
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/574.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/574.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='High Peak Community Arts',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/583.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/583.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='People Express',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0.600000023841858,0,0.600000023841858))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/589.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/589.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='QArts/Studios',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0.600000023841858,0,0.600000023841858))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/593.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/593.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='First Movement',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(1,0,0.200000002980232))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/612.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/612.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='City Arts',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0.600000023841858,0,0.600000023841858))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/615.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/615.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Indigo Dance Group (Salamanda Tandem)',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/623.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/623.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Watering Seeds',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/630.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/630.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Fased In The Arts',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/633.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/633.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='27a Access Artspace',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(1,0,0.200000002980232))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/638.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/638.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Bamboozle Theatre Company',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/648.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/648.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Mantle Community Arts',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/658.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/658.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Artlink East',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/665.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/665.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='Creations',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/670.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/670.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='New Perspectives',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(1,0,0.200000002980232))),
            geometry=Sphere(radius=5.10000002384186))])]),
      Anchor(url=["javascript:window.open(\'https://coderextreme.net/X3DJSONLD/src/main/data/671.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');","javascript:window.open(\'./data/671.html\',\'details\',\'height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no\');"],description='UKan2',
        children=[
        Transform(
          children=[
          Shape(
            appearance=Appearance(
              material=Material(diffuseColor=(1,1,1),emissiveColor=(0,0.300000011920929,1))),
            geometry=Sphere(radius=5.10000002384186))])])])])
) # X3D model complete

###############################################
# Self-test diagnostics
###############################################

if        metaDiagnostics(newModel): # built-in utility method in X3D class
    print(metaDiagnostics(newModel))
# print('check newModel.XML() serialization...')
newModelXML= newModel.XML() # test export method XML() for exceptions during export
newModel.XMLvalidate()

try:
#   print('check newModel.VRML() serialization...')
    newModelVRML=newModel.VRML() # test export method VRML() for exceptions during export
    # print(prependLineNumbers(newModelVRML)) # debug
    print("Python-to-VRML export of VRML output successful")
except BaseException as err:
    print("*** Python-to-VRML export of VRML output failed:", err)
    if newModelVRML: # may have failed to generate
        print(prependLineNumbers(newModelVRML, err.lineno))

try:
#   print('check newModel.JSON() serialization...')
    newModelJSON=newModel.JSON() # test export method JSON() for exceptions during export
#   print(prependLineNumbers(newModelJSON)) # debug
    print("Python-to-JSON export of JSON output successful (still testing)")
except SyntaxError as err:
    print("*** Python-to-JSON export of JSON output failed:", err)
    if newModelJSON: # may have failed to generate
        print(prependLineNumbers(newModelJSON,err.lineno))

print("python x3d.py load and self-test complete for t3.py")

File: src/main/node/net/ecma/data/t3.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var component = require('./x3d.mjs');
var SFInt32 = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var NavigationInfo = require('./x3d.mjs');
var MFFloat = require('./x3d.mjs');
var SFFloat = require('./x3d.mjs');
var WorldInfo = require('./x3d.mjs');
var Viewpoint = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var SFRotation = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var Anchor = require('./x3d.mjs');
var MFString = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var SFColor = require('./x3d.mjs');
var Sphere = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Immersive"),
      version : new SFString("4.0"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("title"),
              content : new SFString("t3.x3d")}),

            new meta({
              name : new SFString("identifier"),
              content : new SFString("http://coderextreme.net/X3DJSONLD/src/main/data/t4.x3d")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("view3dscene, https://castle-engine.io/view3dscene.php")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("DOM2JSONSerializer.js, https://github.com/coderextreme/X3DJSONLD/blob/master/src/main/node/DOM2JSONSerializer.js")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Andreas Plesch and John Carlson")}),

            new meta({
              name : new SFString("source"),
              content : new SFString("t1.wrl")}),

            new meta({
              name : new SFString("description"),
              content : new SFString("Test Case for Proto Expander")}),

            new meta({
              name : new SFString("license"),
              content : new SFString("https://www.web3d.org/x3d/content/examples/license.html")}),

            new meta({
              name : new SFString("translated"),
              content : new SFString("12 May 2020")}),
          component : new SFNode(
            new component({
              name : new SFString("Networking"),
              level : new SFInt32(2)})]),
          component : new SFNode(
            new component({
              name : new SFString("Core"),
              level : new SFInt32(2)}))})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new NavigationInfo({
              type : ["EXAMINE","FLY","WALK"],
              avatarSize : new MFFloat([200,200,120]),
              speed : new SFFloat(3)}),

            new WorldInfo({
              title : new SFString("Arts Mapper")}),

            new Viewpoint({
              description : new SFString("looking North"),
              position : new SFVec3f([0,60,110]),
              orientation : new SFRotation([1,0,0,-0.699999988079071]),
              fieldOfView : new SFFloat(0.7853981)}),

            new Transform({
              translation : new SFVec3f([-468,0,315]),
              children : new MFNode([
                new Anchor({
                  description : new SFString("High Peak Community Arts"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/574.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/574.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("People Express"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/583.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/583.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0.6,0,0.6])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("QArts/Studios"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/589.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/589.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0.6,0,0.6])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("First Movement"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/593.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/593.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([1,0,0.2])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("City Arts"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/612.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/612.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0.6,0,0.6])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Indigo Dance Group (Salamanda Tandem)"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/615.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/615.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Watering Seeds"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/623.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/623.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Fased In The Arts"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/630.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/630.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("27a Access Artspace"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/633.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/633.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([1,0,0.2])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Bamboozle Theatre Company"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/638.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/638.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Mantle Community Arts"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/648.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/648.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Artlink East"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/658.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/658.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("Creations"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/665.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/665.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("New Perspectives"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/670.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/670.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([1,0,0.2])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])}),

                new Anchor({
                  description : new SFString("UKan2"),
                  url : new MFString(["javascript:window.open('https://coderextreme.net/X3DJSONLD/src/main/data/671.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');","javascript:window.open('./data/671.html','details','height=550,width=400,top=50,left=50,menubar=no,status=no,toolbar=no,titlebar=no');"]),
                  children : new MFNode([
                    new Transform({
                      child : new SFNode(
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  diffuseColor : new SFColor([1,1,1]),
                                  emissiveColor : new SFColor([0,0.3,1])}))})),
                          geometry : new SFNode(
                            new Sphere({
                              radius : new SFFloat(5.1)}))}))})])})])})])}))});
console.log(X3D0.toXMLNode());
